### PR TITLE
Task 3

### PR DESCRIPTION
**Pull request by SamoylenkVi**

### CODE REVIEW

File: src/components/add-review-form.jsx [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    const { value } = evt.target;
+    setReview((prevReview) => ({ ...prevReview, commentText: value }));
+  };
+
+  return (
+    <div className="add-review">
+      <form action="#" className="add-review__form">
+        <div className="rating">
+          <div className="rating__stars">
+            <input onChange={handleRatingFilm} className="rating__input" id="star-1" type="radio" name="rating" value="1"/>
```
Comment (Pavel-Liteiniy) [inline src/components/add-review-form.jsx]: Let's make it with using array.map

```suggestion
            {
              RATINGS.map((...)) => (
                <input ... />
              )
            }
```

File: src/components/add-review-form.jsx [hunk 2]
```diff
@@ -0,0 +1,66 @@
+import React, { useState } from 'react';
+
```
Comment (Pavel-Liteiniy) [inline src/components/add-review-form.jsx:5]:
```suggestion
const RATINGS = Array.from({ length: 10 }, (...) => {...})
```

File: src/components/add-review-page.jsx [hunk 3]
```diff
@@ -1,91 +1,59 @@
...
-          <label className="rating__label" htmlFor="star-8">Rating 8</label>
-
-          <input className="rating__input" id="star-9" type="radio" name="rating" value="9" />
-          <label className="rating__label" htmlFor="star-9">Rating 9</label>
-
-          <input className="rating__input" id="star-10" type="radio" name="rating" value="10" />
-          <label className="rating__label" htmlFor="star-10">Rating 10</label>
+import PropTypes from 'prop-types';
+import { Link } from 'react-router-dom';
+import { regexAlt } from '../const';
```
Comment (Pavel-Liteiniy) [inline src/components/add-review-page.jsx]: Lets change file name to `constants`

File: src/components/film-list.jsx [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+import { FILM_CARD_COUNT } from '../const';
+
+export const FilmList = ({ filmsData }) => {
+  const [, setActiveCardId] = useState();
+
+  const handleMouseEnter = (id) => {
+    setActiveCardId(id);
+  };
+
+  return <>
```
Comment (Pavel-Liteiniy) [inline src/components/film-list.jsx]:
```suggestion
  return (
    <>
      ...
    </>
  );
```

File: src/const.js [hunk 5]
```diff
@@ -0,0 +1,2 @@
+export const FILM_CARD_COUNT = 8;
+export const regexAlt = /(?<=img\/).+(?=\.)/gm;
```
Comment (Pavel-Liteiniy) [inline src/const.js]:
```suggestion
export const REGEX_ALT = /(?<=img\/).+(?=\.)/gm;
```

File: src/index.js [hunk 6]
```diff
@@ -21,23 +24,23 @@ const router = createBrowserRouter([
...
+    element: <MyList films={films} />,
   },
   {
     path: '/films/:id',
     element: <MovieDetailsPage />,
   },
   {
     path: '/films/:id/review',
-    element: <AddReviewPage />,
+    element: <AddReviewPage name={film.name} previewImage={film.previewImage}/>,
```
Comment (Pavel-Liteiniy) [inline src/index.js]:
```suggestion
    element: <AddReviewPage />,
```

Comment (Pavel-Liteiniy) [reply src/index.js]: Let's use [useParams](https://v5.reactrouter.com/web/api/Hooks/useparams) inside `AddReviewPage` and MovieDetailsPage too.

File: src/index.js [hunk 7]
```diff
@@ -21,23 +24,23 @@ const router = createBrowserRouter([
...
     path: '/',
-    element: <MainPage />,
+    element: <MainPage films={films} />,
   },
   {
     path: '/login',
     element: <Login />,
   },
   {
     path: '/mylist',
```
Comment (Pavel-Liteiniy) [inline src/index.js]:
```suggestion
    path: '/my-list',
```

File: src/index.js [hunk 8]
```diff
@@ -13,6 +13,9 @@ import { MyList } from './components/my-list.jsx';
 import { MovieDetailsPage } from './components/movie-details-page.jsx';
 import { AddReviewPage } from './components/add-review-page.jsx';
 import { Player } from './components/player.jsx';
+import { films } from './mocks/films';
+
+const film = films[2];
```
Comment (Pavel-Liteiniy) [inline src/index.js]:
```suggestion
```

**Review by Pavel-Liteiniy — CHANGES_REQUESTED** [review]

File: src/components/add-review-page.jsx (hunk 3 above)
Comment (SamoylenkVi) [reply src/components/add-review-page.jsx]: +

File: src/const.js (hunk 5 above)
Comment (SamoylenkVi) [reply src/const.js]: +

File: src/index.js (hunk 7 above)
Comment (SamoylenkVi) [reply src/index.js]: готово

File: src/components/film-list.jsx (hunk 4 above)
Comment (SamoylenkVi) [reply src/components/film-list.jsx]: готово

File: src/index.js (hunk 6 above)
Comment (SamoylenkVi) [reply src/index.js]: готово

File: src/index.js (hunk 8 above)
Comment (SamoylenkVi) [reply src/index.js]: готово

File: src/components/add-review-form.jsx (hunk 1 above)
Comment (SamoylenkVi) [reply src/components/add-review-form.jsx]: готово

File: src/components/add-review-form.jsx (hunk 2 above)
Comment (SamoylenkVi) [reply src/components/add-review-form.jsx:5]: готово

File: src/components/add-review-form.jsx [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+    <div className="add-review">
+      <form action="#" className="add-review__form">
+        <div className="rating">
+          <div className="rating__stars">
+          {
+          ratings.map((rating, index) => (
+            <>
+              <input
+                onChange={handleRatingFilm}
+                key={index}
```
Comment (Pavel-Liteiniy) [inline src/components/add-review-form.jsx]: Using index as key is a bad practice

File: src/components/add-review-form.jsx [hunk 10]
```diff
@@ -0,0 +1,59 @@
...
+  };
+
+  return (
+    <div className="add-review">
+      <form action="#" className="add-review__form">
+        <div className="rating">
+          <div className="rating__stars">
+          {
+          ratings.map((rating, index) => (
+            <>
```
Comment (Pavel-Liteiniy) [inline src/components/add-review-form.jsx]:
```suggestion
            <React.Fragment key={rating}>
```

File: src/components/film-list.jsx [hunk 11]
```diff
@@ -0,0 +1,36 @@
...
+    setActiveCardId(id);
+  };
+
+  return (
+    <>
+      <div className="catalog__movies-list">
+        {Array.from(
+          { length: FILM_CARD_COUNT },
+          (_, index) => <FilmCard
+            key={index}
```
Comment (Pavel-Liteiniy) [inline src/components/film-list.jsx]: bad practise too

Comment (Pavel-Liteiniy) [reply src/components/film-list.jsx]:
```suggestion
            key={filmsData[index].id}
```

File: src/components/my-list.jsx [hunk 12]
```diff
@@ -1,7 +1,12 @@
 import React from 'react';
+import PropTypes from 'prop-types';
 import { Footer } from './footer.jsx';
+import { FilmCard } from './film-card.jsx';
+import { FILM_CARD_COUNT } from '../constants';
 
-export const MyList = () => (
+export const MyList = (props) => {
```
Comment (Pavel-Liteiniy) [inline src/components/my-list.jsx]:
```suggestion
export const MyList = ({ films }) => {
```

File: src/components/my-list.jsx [hunk 13]
```diff
@@ -25,89 +30,21 @@ export const MyList = () => (
...
-            <img src="img/shutter-island.jpg" alt="Shutter Island" width="280" height="175" />
-          </div>
-          <h3 className="small-movie-card__title">
-            <a className="small-movie-card__link" href="movie-page.html">Shutter Island</a>
-          </h3>
-        </article>
+      {Array.from(
+        { length: FILM_CARD_COUNT },
+        (_, index) => <FilmCard
+          key={index}
```
Comment (Pavel-Liteiniy) [inline src/components/my-list.jsx]: bad practice

**Review by Pavel-Liteiniy — CHANGES_REQUESTED** [review]

File: src/components/add-review-form.jsx (hunk 10 above)
Comment (SamoylenkVi) [reply src/components/add-review-form.jsx]: готово

File: src/components/add-review-form.jsx (hunk 9 above)
Comment (SamoylenkVi) [reply src/components/add-review-form.jsx]: готово

File: src/components/film-list.jsx (hunk 11 above)
Comment (SamoylenkVi) [reply src/components/film-list.jsx]: готово

File: src/components/my-list.jsx (hunk 13 above)
Comment (SamoylenkVi) [reply src/components/my-list.jsx]: готво

File: src/components/my-list.jsx (hunk 12 above)
Comment (SamoylenkVi) [reply src/components/my-list.jsx]: готово

**Comment by Pavel-Liteiniy** [comment]
OK

**Pull request merged**